### PR TITLE
push bluetooth-le behind feature flag

### PR DESCRIPTION
**Pull request by PeterGrace**
The introduction of `btleplug` module includes the requirement of libdbus-1-dev which is not available for all cross-compilation platforms.  

This PR ensures that the BLE code is behind a feature flag so that people who want said functionality, can use it.

This PR also updates the protobufs to the latest version in the repository.  I can rebase without this, if necessary.

(cc @lukipuki )

### CODE REVIEW

File: src/utils_internal.rs
Comment (lukipuki) [inline src/utils_internal.rs:2]: Could you use `if cfg!(....) ` instead to have a multi-line condition?
https://doc.rust-lang.org/reference/conditional-compilation.html#the-cfg-macro

**Review by lukipuki — COMMENTED** [review]
I think we should also modify CI to check the build with and without this feature.
This is the workflow file: https://github.com/meshtastic/rust/blob/923d2b7ec5b831a1fd2dc9f05d6d84424acfd3ec/.github/workflows/testing.yml#L40

File: src/utils_internal.rs
Comment (PeterGrace) [reply src/utils_internal.rs:2]: > Could you use `if cfg!(....) ` instead to have a multi-line condition? https://doc.rust-lang.org/reference/conditional-compilation.html#the-cfg-macro

I don't think that's actually possible for use declarations; we could move the use declarations inside of the function that required it, but we'd still run afoul of situations where the elided use was part of the function signature eg

```rust
  #[cfg(feature = "bluetooth-le")]
  async fn scan_peripherals(adapter: &Adapter) -> Result<Vec<Peripheral>, btleplug::Error> {
    use btleplug::platform::{Adapter, Manager, Peripheral};

      adapter
          .start_scan(ScanFilter {
              services: vec![MSH_SERVICE],
          })
          .await?;
      adapter.peripherals().await
  }
```

Comment (lukipuki) [reply src/utils_internal.rs:2]: Sorry for a late response.

It pollutes the code right now but I'd vote for merging as it solves a problem for you. I can have a look for a better solution once I have time in about a month.

**Review by lukipuki — APPROVED** [review]
Approved but I think you need a maintainer's approval, too.

File: src/utils_internal.rs
Comment (PeterGrace) [reply src/utils_internal.rs:2]: If you have a better way to do it please feel free -- I changed my tools to use my fork of this repo so we've got time to come up with an 'optimal solution'

Comment (lukipuki) [reply src/utils_internal.rs:2]: I've actually worked with `#[cfg(...)]` at my job for the last week and while useful, some of its ergonomics is not that good. It's hard to improve on what you've done.

**Pull request merged**
